src/events.py

```diff
@@ -1,4 +1,5 @@
 import math
+from future.utils import with_metaclass
```

Comment:
I personally would be inclined to use [`six`](https://pypi.python.org/pypi/six) instead of [`future`](https://pypi.python.org/pypi/future) -- it seems less intrusive.  Although the actual `with_metaclass` implementation is [just 8 lines of code](https://github.com/benjaminp/six/blob/15e31431af97e5e64b80af0a3f598d382bcdd49a/six.py#L819-L832), so maybe just copy it here to avoid dealing with extra dependencies?